pkg/util/azureclient/azuresdk/armkeyvault/vaults.go

```diff
@@ -0,0 +1,37 @@
+package armkeyvault
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
+)
+
+type VaultsClient interface {
+	CheckNameAvailability(ctx context.Context, vaultName armkeyvault.VaultCheckNameAvailabilityParameters, options *armkeyvault.VaultsClientCheckNameAvailabilityOptions) (armkeyvault.VaultsClientCheckNameAvailabilityResponse, error)
+}
+
+type vaultsClient struct {
+	*armkeyvault.VaultsClient
+}
+
+var _ VaultsClient = &vaultsClient{}
+
+func NewVaultsClient(subscriptionID string, credential azcore.TokenCredential, options *azidentity.EnvironmentCredentialOptions) (VaultsClient, error) {
```

Comment:
I noticed that this track 2 SDK client takes in an `azidentity.EnvironmentCredentialsOption` here rather than an `arm.ClientOptions` like the [subnets client](https://github.com/Azure/ARO-RP/blob/master/pkg/util/azureclient/azuresdk/armnetwork/subnets.go), the first one that was migrated to track 2, does.

I see why you did it this way - because in the context of `pkg/util/cluster`, you are given an `azidentity.EnvironmentCredentialOptions` to work with - but I think all of the track 2 SDK clients should take in the same type for the `options` parameter for consistency's sake.

@cadenmarchese , since you were the one who did that subnet client migration, what do you think?

Reply:
thank you for pointing out. i agree that a consistant interface is important, therefore, i have changed the PR.